rust/src/array.rs

```diff
@@ -200,6 +201,55 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
         &raw[offset as usize..offset as usize + len as usize]
     }
 
+    //TODO: need help here ... 
+
```

Comment:
@paddyhoran @sunchao I wonder if you can help me figure this out. Since the change to use specialization, I don't know how to make this work for numeric types but not boolean types. I'll keep trying but I feel I am missing something.